ssl/rio/poll_immediate.c

```diff
@@ -15,6 +15,20 @@
 #include "../ssl_local.h"
 #include "poll_builder.h"
 
+#if defined(_AIX)
+/*
+ * Some versions of AIX define macros for events and revents for use when
+ * accessing pollfd structures (see Github issue #24236). That interferes
+ * with our use of these names here. We simply undef them.
+ */
+# ifdef revents
+#  undef revents
+# endif
+# ifdef events
+#  undef events
+# endif
```

Comment:
AFAIK there is no need to guard `#undef` with `#ifdef`.

Reply:
This addresses the issue when building openssl itself, but won't any applications that include poll.h on aix suffer from the same build error in the event that they attempt to read or write SSL_POLL_EVENT.[events|revents] ?

Reply:
My comment was not resolved. It is pointless to guard `#undef` with `#ifdef`, it is just a clutter. Just guarding this with `#if defined(_AIX)` is sufficient.

Reply:
> This addresses the issue when building openssl itself, but won't any applications that include poll.h on aix suffer from the same build error in the event that they attempt to read or write SSL_POLL_EVENT.[events|revents] ?

Yes - but only if they include both ssl.h and poll.h. But I don't really see that as our issue to address. The problem occurs because of a bad implementation of the header on AIX. It is not an OpenSSL bug. Similarly workarounds can be done by applications if they need to.